src/tools/rbd_wnbd/rbd_wnbd.cc

```diff
@@ -1159,8 +1195,24 @@ static int do_map(Config *cfg)
     global_init_postfork_finish(g_ceph_context);
   }
 
-  handler->wait();
-  handler->shutdown();
+  {
+    uint64_t watch_handle;
+    WNBDWatchCtx watch_ctx = WNBDWatchCtx(io_ctx, handler, image, info.size);
```

Comment:
Nit: no need to mention the type twice -- just use direct initialization:
```suggestion
    WNBDWatchCtx watch_ctx(io_ctx, handler, image, info.size);
```